### DISCUSSION TITLE
OML 3.0 Scope

### DISCUSSION THREAD

**Discussion by melaasar** [opencaesar/platform#26] [Ideas]
Hey all. I would like to start planning the scope of the next major revision (v3.0) of OML and its tooling ecosystem and I like to get input from you all. So if you like to propose a new feature or an enhancement, please add a comment to this discussion item (optionally referencing a Github issue describing your request further). You can also "upvote" a feature and enhancement proposed by others. This will be a good way for the maintaining team to prioritize the scope of OML 3.0.

**Comment by melaasar** [comment]
[owl-adapter#58](https://github.com/opencaesar/owl-adapter/issues/58) Support OML ontologies importing OWL ontologies directly (i.e., without first converting the OWL to OML).

**Comment by melaasar** [comment]
Support OML text files (*.oml) importing OML XMI (*oml.xmi) and OML JSON (*.omljson) files.

**Comment by melaasar** [comment]
Support a CoPilot (AI Assistant) feature for OML editors as part of the next-gen OML Vision IDE (VS Code extension).

**Comment by melaasar** [comment]
Support simpler import resolution mechanism than XML catalogs, by declaring ontology paths that may be folders or jar files.

**Comment by melaasar** [comment]
Support a .oml.yml file that declares the main settings of an OML project and which can be read by the build script.

**Comment by melaasar** [comment]
Support an OML CLI that helps create new OML projects and perform common editing/analysis tasks from a terminal.

**Reply by SalasCorderoS** [reply]
How would this be different than accessing the project files from the terminal and editing accordingly?

**Reply by melaasar** [reply]
It would provide complementary functions similar to the ones provided by the UI in an IDE. For example, can create new projects, create new ontologies, move ontologies, etc. Other functions like editing ontologies or running gradle tasks can already be done from a CLI.

**Comment by melaasar** [comment]
Improve default ontology diagrams in OML Luxor to be on par or better with those in OML Rosetta

**Reply by SalasCorderoS** [reply]
Perhaps a simple example could reflect better whether this would be needed/wanted

**Comment by melaasar** [comment]
Add the ability to define multiple OML ontologies in the same model file

**Comment by melaasar** [comment]
Add support for Punning (using the same name to denote a class and an individual) to OML

**Reply by NicolasRouquette** [reply]
For an example of punning, see [gufo:categorizes](https://github.com/nemo-ufes/gufo/blob/master/gufo.ttl#L112-L138)
